base/sparse/sparsevector.jl

```diff
@@ -1681,3 +1681,19 @@ droptol!(x::SparseVector, tol, trim::Bool = true) = fkeep!(x, (i, x, tol) -> abs
 
 dropzeros!(x::SparseVector, trim::Bool = true) = fkeep!(x, (i, x, other) -> x != 0, nothing, trim)
 dropzeros(x::SparseVector, trim::Bool = true) = dropzeros!(copy(x), trim)
+
+import Base.fill!
+function fill!(A:: Union{SparseVector, SparseMatrixCSC}, x)
+    # in case of fillvalue 0, only want to iterate over
+    # nonzero slots of A, thats why we have this method
+    T = eltype(A)
+    xT = convert(T, x)
+    if xT == zero(T)
+        fill!(A.nzval, xT)
+    else
+        for i in eachindex(A)
+            @inbounds A[i] = xT
```

Comment:
Yeah but this is only the nonzero case right? And filling a sparse array with nonzeros is an operation which will not be used in code that has to be performant anyway. So I think its not worth optimizing this, though if people here feel different, I may do it.